universal-login-sdk/src/core/services/WalletStorageService.ts

```diff
@@ -22,22 +21,20 @@ export class WalletStorageService implements WalletStorage {
   }
 
   async migrate() {
-    try {
-      const data = this.storageService.get(DEPRECATED_STORAGE_KEY);
-      if (data === null) return;
-      const {wallet} = cast(JSON.parse(data), asObject({
-        kind: asExactly('Deployed'),
-        wallet: asSerializedDeployedWallet,
-      }));
-      for (const network of ['mainnet', 'kovan', 'rinkeby', 'ropsten']) {
-        const providerService = new ProviderService(getProviderForNetwork(network));
-        if (await providerService.isContract(wallet.contractAddress)) {
-          this.storageService.remove(DEPRECATED_STORAGE_KEY);
-          this.storageService.set(`${DEPRECATED_STORAGE_KEY}-${network}`, data);
-          return;
+    for (const network of ['mainnet', 'kovan', 'rinkeby', 'ropsten']) {
+      try {
+        const data = this.storageService.get(`${DEPRECATED_STORAGE_KEY}-${network}`);
+        if (data === null) continue;
+        const state = cast(JSON.parse(data), asObject({
+          kind: asExactly('Deployed'),
+          wallet: asSerializedDeployedWallet,
+        }));
+        if (state?.kind !== 'Deployed') continue;
```

Comment:
We should also return here

Reply:
But I also think that this situation is not possible, because we are using cast a few lines higher. If cast won't get `Deployed` wallet it will throw an error and we catch it.

Reply:
👍 I forgot why I added it, but I think is possible because of typescript. I will check it